modules/yieldmoBidAdapter.js

```diff
@@ -0,0 +1,153 @@
+ * @returns {{callBids: _callBids}}
+ * @constructor
+ */
+
+var YieldmoAdapter = function YieldmoAdapter() {
+  function _callBids(params) {
+    var bids = params.bids;
+    adloader.loadScript(buildYieldmoCall(bids));
+  }
+
+  function buildYieldmoCall(bids) {
+    // build our base tag, based on if we are http or https
+    var ymURI = '//bid.yieldmo.com/exchange/prebid?';
+    var ymCall = document.location.protocol + ymURI;
+
+    // Placement specific information
+    ymCall = _appendPlacementInformation(ymCall, bids);
+
+    // General impression params
+    ymCall = _appendImpressionInformation(ymCall);
+
+    // remove the trailing "&"
+    if (ymCall.lastIndexOf('&') === ymCall.length - 1) {
+      ymCall = ymCall.substring(0, ymCall.length - 1);
+    }
+
+    // @if NODE_ENV='debug'
+    utils.logMessage('ymCall request built: ' + ymCall);
+    // @endif
+
+    return ymCall;
+  }
+
+  function _appendPlacementInformation(url, bids) {
+    var placements = [];
+    var placement;
+    var bid;
+
+    for (var i = 0; i < bids.length; i++) {
+      bid = bids[i];
+
+      placement = {};
+      placement.callback_id = bid.bidId;
+      placement.placement_id = bid.placementCode;
+      placement.sizes = bid.sizes;
+
+      placements.push(placement);
+    }
+
+    url = utils.tryAppendQueryString(url, 'p', JSON.stringify(placements));
+    return url;
+  }
+
+  function _appendImpressionInformation(url) {
+    var page_url = document.location; // page url
+    var pr = document.referrer || ''; // page's referrer
+    var dnt = (navigator.doNotTrack || false).toString(); // true if user enabled dnt (false by default)
+    var _s = document.location.protocol === 'https:' ? 1 : 0; // 1 if page is secure
+    var description = _getPageDescription();
+    var title = document.title || ''; // Value of the title from the publisher's page. 
+    var bust = new Date().getTime().toString(); // cache buster
+    var scrd = window.devicePixelRatio || 0; // screen pixel density
+
+    url = utils.tryAppendQueryString(url, 'callback', '$$PREBID_GLOBAL$$.YMCB');
+    url = utils.tryAppendQueryString(url, 'page_url', page_url);
+    url = utils.tryAppendQueryString(url, 'pr', pr);
+    url = utils.tryAppendQueryString(url, 'bust', bust);
+    url = utils.tryAppendQueryString(url, '_s', _s);
+    url = utils.tryAppendQueryString(url, 'scrd', scrd);
+    url = utils.tryAppendQueryString(url, 'dnt', dnt);
+    url = utils.tryAppendQueryString(url, 'description', description);
+    url = utils.tryAppendQueryString(url, 'title', title);
+
+    return url;
+  }
+
+  function _getPageDescription() {
+    if (document.querySelector('meta[name="description"]')) {
+      return document.querySelector('meta[name="description"]').getAttribute('content'); // Value of the description metadata from the publisher's page.  
+    } else {
+      return '';
+    }
+  }
+
+  // expose the callback to the global object:
+  $$PREBID_GLOBAL$$.YMCB = function(ymResponses) {
```

Comment:
This is fine for right now... but beware that Prebid will be dropping support for JSONP in version 1.0, which should be coming in the next few months.

Reply:
Okay, I'll make a note for us to update this. Do you know if any existing adapters are a good example to look at for 1.0 compatible callback?

Reply:
We won't be letting servers respond with code in 1.0--just data. So you'll be making a GET/POST request, and then unpacking bids from the JSON (or XML, plain text, etc) in the response body.

The exact API for adapters in 1.0 is still under discussion in #1494, if you want to take a look and make suggestions over there. I'm updating the `appnexusAst` adapter to help prototype it.